File: c/hash-table/contains-duplicate-ii.c
bool containsNearbyDuplicate(int* nums, int numsSize, int k) {
    int i,j;
    for(i = 0; i < numsSize; ++i){
        for(j = i + 1; j <= i + k && j < numsSize; ++j)
            if(nums[i] == nums[j])
                return true;
    }

    return false;
}

File: python/hash-table/contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        d = {}
        for i in range(len(nums)):
            idx = d.get(nums[i])
            if idx >= 0 and i - idx <= k:
                return True

            d[nums[i]] = i

        return False
